nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,323 @@
+
+## When to use Custom Fields
+
+Custom fields are commonly used for fields that need different values across individual objects. For example, a custom field on devices to reference an internal ticket number that inventories each device in Nautobot. If you need a solution for marking multiple objects with a common flag, [tags](../models/extras/tag.md) may be a better fit.
+
+## Create a Custom Field
+
+Navigate to the custom fields page by clicking on **Extensibility -> Custom Fields** in the Nautobot menu. Click on **Add** to create a new custom field.
+
+### Custom Field Attributes
+
+#### Label
+
+The label is the human readable label of the custom field that will be displayed on the associated object detail view.
+
+![Custom Field Labels](./images/custom-fields/custom_field_detail_label.png)
+
+#### Grouping
+
+The optional grouping field allows you to group custom fields into collapsible menus.
+
+![Custom Field Grouping](./images/custom-fields/custom_field_detail_grouped.png)
+
+#### Slug
+
+The slug is used to create the URL endpoint for the custom field and is also used as the key in the underlying custom field data dictionary. This is automatically created from the label if not supplied. The default value should be sufficient for most deployments.
+
+!!! tip
+    Because custom field data is included in the database, in the REST API and in GraphQL, we strongly recommend that when defining a custom field, you provide a `slug` that contains underscores rather than dashes (`my_field_slug`, not `my-field-slug`), as some features may not work optimally if dashes are included in the slug. Similarly, the provided `name` should also contain only alphanumeric characters and underscores, as it is currently treated in some cases like a slug.
+
+#### Type
+
+The type of data that the custom field will store. Valid choices are documented in the custom field [model documentation](../models/extras/customfield.md#creating-custom-fields).
+
+#### Weight
+
+Weight determines how custom fields are sorted in forms and object detail views. Higher-weight fields will be ordered lower on the page; if multiple fields have the same weight, they will be listed alphabetically.
+
+#### Description
+
+The description of a custom field is shown as a mouseover tooltip in object detail views and as help text under form fields.
+
+![Custom Field Edit](./images/custom-fields/custom_field_edit.png)
+
+#### Required
+
+Check the required box if this field cannot be null on the associated objects.
+
+!!! warning
+    If an associated object does not have a valid value assigned to a required custom field, that field must be updated with a valid value before the object can be saved. Try to supply a valid default value when creating required custom fields. Since automatic provisioning is only performed when a custom field's content types change, if an existing custom field is changed from optional to required the associated objects will have to be updated manually.
```

Comment:
```suggestion
    If an associated object does not have a valid value assigned to a required custom field, that field must be updated with a valid value before the object can be saved. Try to supply a valid default value when creating required custom fields. Since automatic provisioning is only performed when the assigned content types for a custom field change, if an existing custom field is changed from optional to required the associated objects will need to be updated manually.
```